bots/image-prepare

```diff
@@ -17,50 +16,122 @@
+    parser.add_argument('-f', '--force', action='store_true', help='Force update of images')
+    parser.add_argument('-b', '--build-image', action='store', help='Build in this image')
+    parser.add_argument('-B', '--build-only', action='store_true', help='Only build and download results')
+    parser.add_argument('-I', '--install-only', action='store_true', help='Only upload and install')
+    parser.add_argument('-c', '--containers', action='store_true', help='Install container images')
+    parser.add_argument('--address', help='Address of already running machine')
+    parser.add_argument('image', nargs='?', default=testvm.DEFAULT_IMAGE, help='The image to use')
+    args = parser.parse_args()
+
+    if args.build_only:
+        install_image = None
+    else:
+        install_image = args.image
+
+    build_image = args.build_image
+    if args.install_only:
+        build_image = None
+    elif not build_image:
+        build_image = get_build_image(install_image or args.image)
+
+    # Default to putting build output in bots directory
+    if "TEST_ATTACHMENTS" in os.environ:
+        results = os.path.join(os.environ["TEST_ATTACHMENTS"], "build-results")
+    else:
+        results = os.path.join(BOTS, "build-results")
+
+    try:
+        if build_image:
+            subprocess.check_call([ "image-download", build_image])
+        if install_image:
+            subprocess.check_call([ "image-download", install_image])
+    except OSError, ex:
+        if ex.errno != errno.ENOENT:
+            raise
+        sys.stderr.write("image-prepare: missing tools to download images\n")
+    except subprocess.CalledProcessError as e:
+        print "unable to download all necessary images", e
+        return 1
+
+    try:
+        build_and_install(install_image, build_image, results, args={
+            "verbose": args.verbose,
+            "sit": args.sit,
+            "quick": args.quick,
+            "build_image": build_image,
+            "build_only": args.build_only,
+            "install_only": args.install_only,
+            "containers": args.containers,
+            "address": args.address,
+        })
+    except RuntimeError as ex:
+        sys.stderr.write("image-prepare: {0}\n".format(str(ex)))
+        return 2
+
+    return 0
 
 def upload_scripts(machine, args):
     machine.execute("rm -rf /var/lib/testvm")
-    machine.upload([ os.path.join(testvm.TEST_DIR, "images", "scripts", "lib") ], "/var/lib/testvm")
-    machine.upload([ os.path.join(testvm.TEST_DIR, "images", "scripts", "%s.install" % machine.image) ], "/var/tmp")
-    machine.upload([os.path.join(testvm.TEST_DIR,"..", "containers")], "/var/tmp")
+    machine.upload([ os.path.join(BOTS, "images", "scripts", "lib") ], "/var/lib/testvm")
+    machine.upload([ os.path.join(BOTS, "images", "scripts", "%s.install" % machine.image) ], "/var/tmp")
+    machine.upload([os.path.join(BASE, "containers")], "/var/tmp")
```

Comment:
nitpick: if we're changing this line anyway, we might as well add spaces to be consistent with the previous lines